src/main/java/net/dries007/tfc/compat/wailatop/WailaBlocks.java

```diff
@@ -0,0 +1,483 @@
+import net.dries007.tfc.objects.items.rock.ItemRock;
+import net.dries007.tfc.objects.te.*;
+import net.dries007.tfc.util.calendar.CalendarTFC;
+import net.dries007.tfc.util.calendar.Month;
+import net.dries007.tfc.util.climate.ClimateTFC;
+import net.dries007.tfc.world.classic.chunkdata.ChunkDataTFC;
+
+
+@ParametersAreNonnullByDefault
+@MethodsReturnNonnullByDefault
+public final class WailaBlocks implements IWailaDataProvider
+{
+    public static void callbackRegister(IWailaRegistrar registrar)
+    {
+
+        registrar.addConfig("TFC", "tfc.displayTemp", true);
+        registrar.addConfig("TFC", "tfc.newtotfc", true);
+
+
+        WailaBlocks dataProvider = new WailaBlocks();
+
+        //Stack
+        registrar.registerStackProvider(dataProvider, BlockOreTFC.class);
+        registrar.registerStackProvider(dataProvider, BlockCropTFC.class);
+        //Head
+        registrar.registerHeadProvider(dataProvider, BlockOreTFC.class);
+        registrar.registerHeadProvider(dataProvider, BlockBarrel.class);
+        registrar.registerHeadProvider(dataProvider, TECropBase.class);
+        registrar.registerHeadProvider(dataProvider, BlockCropDead.class);
+        registrar.registerHeadProvider(dataProvider, BlockFruitTreeLeaves.class);
+        registrar.registerHeadProvider(dataProvider, BlockFruitTreeTrunk.class);
+        registrar.registerHeadProvider(dataProvider, BlockFruitTreeBranch.class);
+
+        //Body
+        registrar.registerBodyProvider(dataProvider, BlockOreTFC.class);
+        registrar.registerBodyProvider(dataProvider, TEPitKiln.class);
+        registrar.registerBodyProvider(dataProvider, TEFirePit.class);
+        registrar.registerBodyProvider(dataProvider, TEBloomery.class);
+        registrar.registerBodyProvider(dataProvider, TEBlastFurnace.class);
+        registrar.registerBodyProvider(dataProvider, TECharcoalForge.class);
+        registrar.registerBodyProvider(dataProvider, TEBarrel.class);
+        registrar.registerBodyProvider(dataProvider, TELogPile.class);
+        registrar.registerBodyProvider(dataProvider, TEPlacedItemFlat.class);
+        registrar.registerBodyProvider(dataProvider, TECropBase.class);
+        registrar.registerBodyProvider(dataProvider, BlockFruitTreeLeaves.class);
+        registrar.registerBodyProvider(dataProvider, BlockBerryBush.class);
+        registrar.registerBodyProvider(dataProvider, BlockCropDead.class);
+
+        //Tail
+        registrar.registerTailProvider(dataProvider, BlockRockVariant.class);
+
+
+    }
+
+    @Override
+    public ItemStack getWailaStack(IWailaDataAccessor accessor, IWailaConfigHandler config)
+    {
+        Block b = accessor.getBlock();
+        ItemStack stack = ItemStack.EMPTY;
+
+        if (b instanceof BlockOreTFC)
+        {
+            stack = getOreTFCStack(accessor, config);
+        }
+        else if (b instanceof BlockCropSimple)
+        {
+            stack = getCropSimpleStack(accessor, config);
+        }
+        return stack;
```

Comment:
If you do the other solution i mentioned it also applies here: You can code in that new interface a `getWailaStack()` there and instead of a lot of if-else, a for loop.